tests/testsuite/freshness.rs

```diff
@@ -1524,18 +1612,20 @@ fn reuse_panic_pm() {
     // bar is built once without panic (for proc-macro) and once with (for the
     // normal dependency).
     p.cargo("build -v")
-        .with_stderr_unordered(
-            "\
+        .with_stderr_data(
+            str![[r#"
 [LOCKING] 3 packages to latest compatible versions
-[COMPILING] bar [..]
-[RUNNING] `rustc --crate-name bar --edition=2015 bar/src/lib.rs [..]--crate-type lib --emit=[..]link[..]
-[RUNNING] `rustc --crate-name bar --edition=2015 bar/src/lib.rs [..]--crate-type lib --emit=[..]link -C panic=abort[..]-C debuginfo=2 [..]
-[COMPILING] somepm [..]
-[RUNNING] `rustc --crate-name somepm [..]
-[COMPILING] foo [..]
-[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs [..]-C panic=abort[..]
-[FINISHED] [..]
-",
+[COMPILING] bar v0.0.1 ([ROOT]/foo/bar)
+[RUNNING] `rustc --crate-name bar [..] bar/src/lib.rs [..]--crate-type lib --emit=[..]link[..]
+[RUNNING] `rustc --crate-name bar [..] bar/src/lib.rs [..]--crate-type lib --emit=[..]link -C panic=abort[..]-C debuginfo=2 [..]
+[COMPILING] somepm v0.0.1 ([ROOT]/foo/somepm)
+[RUNNING] `rustc --crate-name somepm [..]`
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo [..]-C panic=abort[..]`
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]]
```

Comment:
I'm trying to fix this `reuse_panic_pm` [CI failure](https://github.com/rust-lang/cargo/actions/runs/9724974056/job/26841764339#step:11:4052), which passed on `Tests on macOS aarch64 stable` but failed on `Test on macOS x86_64 stable` with below:

```diff

---- expected: tests/testsuite/freshness.rs:1616:13
++++ actual:   stderr
   1    1 | [LOCKING] 3 packages to latest compatible versions
   2    2 | [COMPILING] bar v0.0.1 ([ROOT]/foo/bar)
   3    3 | [RUNNING] `rustc --crate-name bar [..] bar/src/lib.rs [..]--crate-type lib --emit=[..]link[..]
-  4      - [RUNNING] `rustc --crate-name bar [..] bar/src/lib.rs [..]--crate-type lib --emit=[..]link -C panic=abort[..]-C debuginfo=2 [..]
   5    4 | [COMPILING] somepm v0.0.1 ([ROOT]/foo/somepm)
   6    5 | [RUNNING] `rustc --crate-name somepm [..]`
   7    6 | [COMPILING] foo v0.0.1 ([ROOT]/foo)
   8    7 | [RUNNING] `rustc --crate-name foo [..]-C panic=abort[..]`
   9    8 | [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+       9 + [RUNNING] `rustc --crate-name bar --edition=2015 bar/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no --check-cfg 'cfg(docsrs)' --check-cfg 'cfg(feature, values())' -C metadata=9229e8e86d572a0d -C extra-filename=-9229e8e86d572a0d --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps`
```

Some thought & findings so far:

Since the test logic was unchanged, I think the arch difference can be ruled out first.

Me next suspect is flakiness, which leads me to this `with_stderr_unordered` caveat from its nice comment:

>     /// Be careful when using patterns such as `[..]`, because you may end up
>     /// with multiple lines that might match, and this is not smart enough to
>     /// do anything like longest-match. For example, avoid something like:
>     ///
>     /// ```text
>     ///  [RUNNING] `rustc [..]
>     ///  [RUNNING] `rustc --crate-name foo [..]
>     /// ```
>     ///
>     /// This will randomly fail if the other crate name is `bar`, and the
>     /// order changes.

The original L1531-L1532 seems to already match this scenario (A line matching L1532 could also match L1531). However, I assume `reuse_panic_pm` has been stable until this something introduced in my PR.

According to the comment form test:
> bar is built once without panic (for proc-macro) and once with (for the normal dependency).

If the order of two `bar` build is non-deterministic this could be a problem; In contrast, if the order is deterministic, that makes me wonder why using `unordered` [in the first place](https://github.com/rust-lang/cargo/commit/86489946d0828af1617ba921d3cc16146a9e0e53#diff-1d79c710a0473fc4ffb021e2bd34a3cdd5a2431a0a9dc905baba7458090064b1R1166-R1176).

In fact, I removed `.unordered`, reran and passed `reuse_panic_pm` a few times locally, so let my try this first.

I also tried to reproduce any randomness introduce by [Snapbox `unordered`], but no finding:
<details>

```rust
use snapbox::prelude::*;
use snapbox::str;
use snapbox::assert_data_eq;

#[cargo_test]
fn test_snapbox_unorder_randomness() {
    let actual = str![[r#"
rustc --crate-name foo --a=b --ccc=ddd
rustc --crate-name foo --a=b --panic=abort --ccc=ddd
eee fff
"#]]
    .unordered();
    let expected = str![[r#"
rustc --crate-name foo [..]
rustc --crate-name foo [..] --panic=abort [..]
eee fff
"#]]
    .unordered();
    assert_data_eq!(actual, expected); // Always pass. Not flaky.
}
```

</details>

P.s. For reference, the full output from my local runs are like:
```
[LOCKING] 3 packages to latest compatible versions
[COMPILING] bar v0.0.1 ([ROOT]/foo/bar)
[RUNNING] `rustc --crate-name bar --edition=2015 bar/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no --check-cfg 'cfg(docsrs)' --check-cfg 'cfg(feature, values())' -C metadata=3eb56625d2059feb -C extra-filename=-3eb56625d2059feb --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps`
[RUNNING] `rustc --crate-name bar --edition=2015 bar/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C panic=abort -C embed-bitcode=no -C debuginfo=2 -C split-debuginfo=unpacked --check-cfg 'cfg(docsrs)' --check-cfg 'cfg(feature, values())' -C metadata=e34f62847e3b5fb0 -C extra-filename=-e34f62847e3b5fb0 --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps`
[COMPILING] somepm v0.0.1 ([ROOT]/foo/somepm)
[RUNNING] `rustc --crate-name somepm --edition=2015 somepm/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type proc-macro --emit=dep-info,link -C prefer-dynamic -C embed-bitcode=no --check-cfg 'cfg(docsrs)' --check-cfg 'cfg(feature, values())' -C metadata=a7ef9925b27e01b6 -C extra-filename=-a7ef9925b27e01b6 --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps --extern bar=[ROOT]/foo/target/debug/deps/libbar-[HASH].rlib --extern proc_macro`
[COMPILING] foo v0.0.1 ([ROOT]/foo)
[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C panic=abort -C embed-bitcode=no -C debuginfo=2 -C split-debuginfo=unpacked --check-cfg 'cfg(docsrs)' --check-cfg 'cfg(feature, values())' -C metadata=ebe9fb0a09bbfc5f -C extra-filename=-ebe9fb0a09bbfc5f --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps --extern bar=[ROOT]/foo/target/debug/deps/libbar-[HASH].rmeta --extern somepm=[ROOT]/foo/target/debug/deps/libsomepm-[HASH].dylib`
[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
```

I guess some of my questions are:
- Just in case, was there any similar `reuse_panic_pm` flakiness you recall seeing prior to this PR?
- Does `reuse_panic_pm` really need `with_stderr_unordered` or `.unordered()`?
- Anything else I should look into?

[Snapbox `unordered`]: https://doc.rust-lang.org/nightly/nightly-rustc/cargo_test_support/prelude/trait.IntoData.html#method.unordered

Reply:
Update: a453efd CI passes `reuse_panic_pm` on all stable pipeline: [linux](https://github.com/rust-lang/cargo/actions/runs/9729603770/job/26851692382#step:11:2249), [mac x86](https://github.com/rust-lang/cargo/actions/runs/9729603770/job/26851692839#step:11:2246), [mac aarch64](https://github.com/rust-lang/cargo/actions/runs/9729603770/job/26851693000#step:11:2245), [Windows](https://github.com/rust-lang/cargo/actions/runs/9729603770/job/26851693071#step:11:2270)

Reply:
Update: The same error happened again after a453efd on `Tests Windows x86_64 gnu nightly`, so a453efd didn't fix the problem 😞 

https://github.com/rust-lang/cargo/actions/runs/9747539766/job/26900484318#step:11:4053

This seems like a flaky error. Dropping a453efd for now.

Reply:
Might need a deeper investigation of it. Fine with holding off for now.